pycsghub/repository.py

```diff
@@ -0,0 +1,278 @@
+import os
+from typing import Optional
+import subprocess
+from typing import List, Optional, Union
+from pathlib import Path
+import requests
+import base64
+import shutil
+import re
+from urllib.parse import urlparse
+from pycsghub.constants import GIT_ATTRIBUTES_CONTENT, REPO_TYPE_DATASET
+from pycsghub.utils import (build_csg_headers,
+                            model_id_to_group_owner_name,
+                            get_endpoint)
+
+class Repository:
+    def __init__(
+        self,
+        repo_id: str,
+        upload_path: str,
+        branch_name: Optional[str] = "main",
+        work_dir: Optional[str] = "/tmp/csg",
+        user_name: Optional[str] = "",
+        token: Optional[str] = "",
+        license: Optional[str] = "apache-2.0",
+        nickname: Optional[str] = "",
+        description: Optional[str] = "",
+        repo_type: Optional[str] = None,
+        endpoint: Optional[str] = None,
+        auto_create: Optional[bool] = True,
+        copy_files: Optional[bool] = True,
+    ):    
+        self.repo_id = repo_id
+        self.upload_path = upload_path
+        self.branch_name = branch_name
+        self.work_dir = work_dir
+        self.user_name = user_name
+        self.token = token
+        self.license = license
+        self.nickname = nickname
+        self.description = description
+        self.repo_type = repo_type
+        self.endpoint = endpoint
+        self.auto_create = auto_create
+        self.copy_files = copy_files
+        if self.repo_type == REPO_TYPE_DATASET:
+            self.repo_url_prefix = "datasets"
+        else:
+            self.repo_url_prefix = "models"
```

Comment:
why not support other repo types like codes and spaces

Reply:
add repo type space and code